torchaudio/transforms.py

```diff
@@ -729,26 +729,24 @@ def __init__(self,
     def forward(self, complex_specgrams: Tensor, overriding_rate: Optional[float] = None) -> Tensor:
         r"""
         Args:
-            complex_specgrams (Tensor): complex spectrogram (..., freq, time, complex=2).
+            complex_specgrams (Tensor):
+                Either a real tensor of dimension of ``(..., freq, num_frame, complex=2)``
+                or a tensor of dimension ``(..., freq, num_frame)`` with complex dtype.
             overriding_rate (float or None, optional): speed up to apply to this batch.
                 If no rate is passed, use ``self.fixed_rate``. (Default: ``None``)
 
         Returns:
-            Tensor: Stretched complex spectrogram of dimension (..., freq, ceil(time/rate), complex=2).
+            Tensor:
+                Stretched spectrogram. The resulting tensor is of the same dtype as the input
+                spectrogram, but the number of frames is changed to ``ceil(num_frame / rate)``.
         """
-        assert complex_specgrams.size(-1) == 2, "complex_specgrams should be a complex tensor, shape (..., complex=2)"
-
         if overriding_rate is None:
+            if self.fixed_rate is None:
+                raise ValueError(
+                    "If no fixed_rate is specified, must pass a valid rate to the forward method.")
             rate = self.fixed_rate
-            if rate is None:
-                raise ValueError("If no fixed_rate is specified"
-                                 ", must pass a valid rate to the forward method.")
         else:
             rate = overriding_rate
-
-        if rate == 1.0:
```

Comment:
could you clarify this change?

Reply:
`TimeStretch` is a thin wrapper around `phase_vocoder`. and not altering the input Tensor when `rate == 1.0` is a valid branching for `phase_vocoder` itself, not just `TimeStretch`. so it makes more sense if it's in `phase_vocoder`.